src/components/PostInFeed.tsx

```diff
@@ -1,38 +1,34 @@
-import {Box, Text, Avatar, Skeleton, HStack, FavouriteIcon, View} from 'native-base';
+import {Box, Text, Skeleton, HStack, FavouriteIcon, Pressable} from 'native-base';
 import { Post } from '../xplat/types/post';
 import {useEffect, useState} from 'react';
 import placeholder_image from '../placeholder_image.jpg';
 import './css/feed.css';
+import { User } from '../xplat/types/user';
+import AuthorHandle from './AuthorHandle';
 
 const PostInFeed = ({post}: {post: Post}) => {
     const [body, setBody] = useState('');
-    const [author, setAuthor] = useState('');
+    const [author, setAuthor] = useState<User>();
     const [hasImage, setHasImage] = useState(false);
-    const [imageURL, setImageURL] = useState<string | undefined>();
+    const [imageURLs, setImageURLs] = useState<(string | undefined)[]>();
     const [isLoaded, setIsLoaded] = useState(false);
-    const [authorAvatar, setAuthorAvatar] = useState('');
     const [time, setTime] = useState<Date>();
+    const [timeString, setTimeString] = useState('');
     const [likes, setLikes] = useState(0);
 
     async function resolvePost() {
-        let _body = post.getTextContent(), _author = post.getAuthor(), _hasImage = post.hasImageContent(),
-            _imageURL = post.getImageContentUrl(), _time = post.getTimestamp(), _likes = post.getLikes();
+        await post.getData();
 
-        await Promise.all([_body, _author, _hasImage, _imageURL, _likes, _time]);
```

Comment:
is this what is causing our components to asynchronously load each field step by step and reupdate each time? Looks like the old way would await every download to take place then update 1 final time

Reply:
The problem with the old way was that it would call each of those in parallel, possibly causing us to fetch from Firebase 7 times rather than just once to initialize the document. I think the best approach to fix the shimmering is to move away from using the isLoaded boolean as a universal load checker and instead for each Skeleton wrapper check whether the component it is wrapping is valid. That's what Mathis said he's doing for mobile